src/rpc/handlers/BookChanges.cpp

```diff
@@ -24,14 +24,15 @@ struct BookChange
 
 class BookChangesHandler
```

Comment:
What is the reason we want this to be a class? As far as I can tell, this is only used in `computeBookChanges`, where it is constructed locally and then immediately destroyed. It doesn't actually store any internal state, and I can't see a reason we would ever want to extend this class. The only argument to its constructor is the backend, which is a singleton inside the application. So all possible instances of `BookChangesHandler` are functionally identical. I feel this class is just a wrapper around the member functions, but the member functions could just be free functions. I feel we are adding a class here just for the sake of adding a class, when free functions would probably be simpler, achieve the same result, and be more appropriate for the problem. But its up to you, that's just my two cents.

Reply:
Initially the design was more fitting than it became now but in general it's for encapsulation and readability. It does store internal state - `offerCancel_` and `tally_` some of which are accessed by different member functions (private ones so that they are defo not called by someone else, like a free function would be able to).
 
But you are not wrong thinking this "is just a wrapper around member functions" - it is. This is better than 5 free functions or one giant function in most cases imo.
I don't have a problem with free function where they make more sense than wrappers/classes/encapsulation but i don't think this is the right use case for them. 

C++ offers us a lot of flexibility and tools in the language - IMHO, we should use them :)

Reply:
I'm not a huge fan of passing "state" around as arguments so what i did instead is i kept the temporary instance approach but i hid it behind a static function - this way we can't misuse the interface (by calling handle on the instance twice for example) and also the correct usage pattern is now obvious as there is now only one way to access it. As a bonus i was able to get rid of the `reset` function because now the implementation is a guaranteed `one-shot`.

I think i quite like this approach, thanks for giving me ideas :+1: